File: raincoat-fabric/src/main/java/dev/uten2c/raincoat/RaincoatModMenuApiImpl.java
package dev.uten2c.raincoat;

import com.terraformersmc.modmenu.api.ConfigScreenFactory;
import com.terraformersmc.modmenu.api.ModMenuApi;
import dev.uten2c.raincoat.screen.RaincoatOptionsScreen;
import net.minecraft.client.MinecraftClient;

public class RaincoatModMenuApiImpl implements ModMenuApi {
    @Override
    public ConfigScreenFactory<?> getModConfigScreenFactory() {
        return parent -> new RaincoatOptionsScreen(parent, MinecraftClient.getInstance().options);
    }
}

File: raincoat-fabric/src/main/kotlin/dev/uten2c/raincoat/modmenu/RaincoatModMenuApiImpl.kt
package dev.uten2c.raincoat.modmenu

import com.terraformersmc.modmenu.api.ConfigScreenFactory
import com.terraformersmc.modmenu.api.ModMenuApi
import dev.uten2c.raincoat.option.RaincoatOptionsScreen
import net.minecraft.client.MinecraftClient

class RaincoatModMenuApiImpl : ModMenuApi {
    override fun getModConfigScreenFactory(): ConfigScreenFactory<*> {
        return ConfigScreenFactory {
            RaincoatOptionsScreen(it, MinecraftClient.getInstance().options)
        }
    }
}
